win32ss/user/ntuser/misc.c

```diff
@@ -308,8 +308,7 @@ NtUserGetThreadState(
          ret = (gpqForeground == GetW32ThreadInfo()->MessageQueue);
          break;
       case THREADSTATE_GETCURSOR:
-         ret = (DWORD_PTR) (GetW32ThreadInfo()->MessageQueue->CursorObject ?
-                            UserHMGetHandle(GetW32ThreadInfo()->MessageQueue->CursorObject) : 0);
+         ret = (DWORD_PTR)UserHMGetHandleSafe(GetW32ThreadInfo()->MessageQueue->CursorObject);
```

Comment:
This code will call `GetW32ThreadInfo` twice.

Reply:
Can inline function fix this?

Reply:
Using the old macro will fix it.

Reply:
Fixed.

Reply:
This cannot be inlined because UserHMGetHandle can be used with any object type (PVOID).

Reply:
Note that in case this would be a problem, a fix could be:
```c
case THREADSTATE_GETCURSOR:
{
    PCURICON_OBJECT CursorObject = GetW32ThreadInfo()->MessageQueue->CursorObject;
    ret = (DWORD_PTR)UserHMGetHandleSafe(CursorObject);
    break;
}
```

Reply:
Fixed,